packages/overlay/src/OverlayTrigger.ts

```diff
@@ -251,26 +256,27 @@ export class OverlayTrigger extends LitElement {
         );
     }
 
-    private async onTriggerLongpress(
-        event?: CustomEvent<LongpressEvent>
-    ): Promise<void> {
+    private _longpressEvent?: CustomEvent<LongpressEvent>;
+
+    private async onTriggerLongpress(): Promise<void> {
         if (!this.targetContent || !this.longpressContent) {
             return;
         }
         const { targetContent, longpressContent } = this;
         this.prepareToFocusOverlayContent(longpressContent);
-        const type =
-            event && event.detail.source === 'keyboard' ? 'click' : 'longpress';
-        const interaction = this.type ? this.type : type || 'longpress';
+        const notImmediatelyClosable =
+            this._longpressEvent?.detail.source !== 'keyboard';
         this.closeLongpressOverlay = await OverlayTrigger.openOverlay(
             targetContent,
-            interaction,
+            this.type ? this.type : 'longpress',
             longpressContent,
             {
                 ...this.overlayOptions,
                 receivesFocus: 'auto',
+                notImmediatelyClosable,
```

Comment:
Expand the state available to the overlay so that `longpress` never needs to be overridden as `click`.